lib/ml/feature_label.hpp

```diff
@@ -14,55 +14,29 @@
 
 #pragma once
 
-#include <cassert>
-#include <cmath>
-#include <string>
-#include <utility>
-#include <vector>
-
-#include "boost/tokenizer.hpp"
-#include "core/engine.hpp"
+#include "lib/vector.hpp"
 
 namespace husky {
 namespace lib {
 namespace ml {
-using vec_double = std::vector<double>;
 
-template <typename FT, typename LT>
-class FeatureLabelBase {
+// extends LabeledPoint, using Vector to represent features
+template <typename FeatureT, typename LabelT, bool is_sparse>
+class FeatureLabel : public LabeledPoint<Vector<FeatureT, is_sparse>, LabelT> {
```

Comment:
Yes. The name need to be decided first or later many files need modification upon renaming it.
Since FeatureLabel, as kelvin says, is a general extension, a long name will result in lengthy subclass names. Is it ok to abbreviate the prefix and use the name LPSomeSuffix?